core/src/impl/Kokkos_BitOps.hpp

```diff
@@ -95,16 +95,17 @@ inline int int_log2_host(unsigned i) {
 #endif
 }
 
-#if defined(__EDG__)
-#pragma push
+#if defined(__EDG__) && !defined(KOKKOS_COMPILER_INTEL)
+#pragma push && !defined(KOKKOS_COMPILER_INTEL)
 #pragma diag_suppress implicit_return_from_non_void_function
 #endif
 KOKKOS_FORCEINLINE_FUNCTION
 int int_log2(unsigned i) {
   KOKKOS_IF_ON_DEVICE((return int_log2_device(i);))
   KOKKOS_IF_ON_HOST((return int_log2_host(i);))
 }
-#if defined(__EDG__)
+#if defined(__EDG__) && !defined(KOKKOS_COMPILER_INTEL) && \
+    !defined(KOKKOS_COMPILER_INTEL)
```

Comment:
Why twice?

Reply:
@crtrott clang-format error, not sure how to prevent it without breaking the clang-format check, any suggestions?

Reply:
Fixed now